pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -652,6 +652,29 @@ public Response endReplaceSegments(
     }
   }
 
+  @GET
+  @Path("segments/{tableName}/listSegmentLineage")
+  @Authenticate(AccessType.READ)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "List segment lineage", notes = "List segment lineage")
+  public Response listSegmentLineage(
```

Comment:
could we add some test for this endpoint. you can checkout how other endpoints are tested under `PinotSegmentUploadDownloadRestletResourceTest`

Reply:
Raised some concerns in the `#Testing done` section of the PR description. Namely, `PinotSegmentUploadDownloadRestletResourceTest` does not seem to cover any segment operations integration tests (as it requires table creation and segment upload) and such operations are tested in `PinotHelixResourceManagerTest` though.

Reply:
oh. actually my mistake. I think the first thing I suggest is to move the API from `PinotSegmentUploadDownloadRestletResource` to `PinotSegmentRestletResource` where all the other segment APIs are located. 
then it should be easy to add a test in `PinotSegmentRestletResourceTest`

Reply:
CC ^ @snleee what do you think? this IMO should be in SegmentRestletResource rather than the UploadDownload one

Reply:
I think that @yuanbenson probably put the new API to `PinotSegmentUploadDownloadRestletResource` because the segment lineage-related code is being written in the segment replacement-related APIs. 

Another way to put this is  `e.g. GET /tables/{tableName}/segments/lineage` under `PinotSegmentRestletResource` as @walterddr suggested.

Reply:
@walterddr Moved the newly added API to `PinotSegmentRestletResource`.